sway-lib-std/src/u128.sw

```diff
@@ -104,6 +115,22 @@ impl U128 {
 
     /// Safely downcast to `u64` without loss of precision.
     /// Returns Err if the number > u64::max()
+    ///
+    /// ### Examples
+    /// 
+    /// ```sway
+    /// use std::u128::{U128, U128Error};
+    ///
+    /// let zero_u128 = U128 { upper: 0, lower: 0 };
+    /// let zero_u64 = zero_u128.as_u64().unwrap();
+    ///
+    /// assert(zero_u64 == 0);
+    /// 
+    /// let max_u128 = U128::max();
+    /// let result = max_u128.as_u64();
+    ///
+    /// assert(result == Result::Err(U128Error::LossOfPrecision));
```

Comment:
Yeah same as for `Option`.. I'm not sure the comparison here works. You could just do:
```suggestion
    /// assert(result.is_err()));
```
Or you'd have to use a `match` statement.